src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -98,83 +98,96 @@ public static int MainInternal(string[] args, CommandOutputLogger log)
-            {
-                bool handled = false;
-                string verb = null;
-                if (args.Length > 1)
-                {
-                    // Redirect users nicely if they do 'dotnet nuget sources add' or 'dotnet nuget add sources'
-                    if (StringComparer.OrdinalIgnoreCase.Compare(args[0], "sources") == 0)
-                    {
-                        verb = args[1];
-                    }
-                    else if (StringComparer.OrdinalIgnoreCase.Compare(args[1], "sources") == 0)
+                    bool handled = HandleCommandLineHelp(args, log);
+                    if (!handled)
                     {
-                        verb = args[0];
+                        LogException(e, log);
+                        ShowBestHelp(app, args);
                     }
 
-                    if (verb != null)
-                    {
-                        switch (verb.ToLowerInvariant())
-                        {
-                            case "add":
-                            case "remove":
-                            case "update":
-                            case "enable":
-                            case "disable":
-                            case "list":
-                                log.LogMinimal(string.Format(CultureInfo.CurrentCulture,
-                                    Strings.Sources_Redirect, $"dotnet nuget {verb} source"));
-                                handled = true;
-                                break;
-                            default:
-                                break;
-                        }
-                    }
+                    exitCode = 1;
                 }
-
-                if (!handled)
+                finally
                 {
-                    // Log the error
-                    if (ExceptionLogger.Instance.ShowStack)
+                    // Limit the exit code range to 0-255 to support POSIX
+                    if (exitCode < 0 || exitCode > 255)
                     {
-                        log.LogError(e.ToString());
-                    }
-                    else
-                    {
-                        log.LogError(ExceptionUtilities.DisplayMessage(e));
+                        exitCode = 1;
                     }
+                }
+            }
+            else
+            {
+                // Run with System.CommandLine
+                exitCode = command.Invoke(args);
+            }
+
+            return exitCode;
+        }
 
-                    // Log the stack trace as verbose output.
-                    log.LogVerbose(e.ToString());
+        private static bool HandleCommandLineHelp(string[] args, CommandOutputLogger log)
+        {
+            string verb = null;
+            bool handled = false;
 
-                    exitCode = 1;
+            if (args.Length > 1)
+            {
+                // Redirect users nicely if they do 'dotnet nuget sources add' or 'dotnet nuget add sources'
```

Comment:
this brought up the discussion of the transition plan between the two command structures. You said you'll talk to dotnet team about their preferences here.

Reply:
Following up in dotnet sdk issue: https://github.com/dotnet/sdk/issues/29480

Reply:
From https://github.com/dotnet/sdk/issues/29480#issuecomment-1347426073

Update: Per spec, `dotnet add source` was preferred to be consistent with .NET CLI in version 1.x
https://github.com/NuGet/Home/wiki/Add-nuget-sources-command-to-the-dotnet-CLI

Then, I will try to align to .NET 7 noun-first approach:

- `dotnet nuget sources add`: Run command
- `dotnet nuget add source`: Show warning message, run command